packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -309,53 +321,48 @@ class StandardMessageCodec implements MessageCodec<dynamic> {
     } else if (value is String) {
       buffer.putUint8(_kString);
       final List<int> bytes = UTF8.encoder.convert(value);
-      _writeSize(buffer, bytes.length);
+      writeSize(buffer, bytes.length);
       buffer.putUint8List(bytes);
     } else if (value is Uint8List) {
       buffer.putUint8(_kUint8List);
-      _writeSize(buffer, value.length);
+      writeSize(buffer, value.length);
       buffer.putUint8List(value);
     } else if (value is Int32List) {
       buffer.putUint8(_kInt32List);
-      _writeSize(buffer, value.length);
+      writeSize(buffer, value.length);
       buffer.putInt32List(value);
     } else if (value is Int64List) {
       buffer.putUint8(_kInt64List);
-      _writeSize(buffer, value.length);
+      writeSize(buffer, value.length);
       buffer.putInt64List(value);
     } else if (value is Float64List) {
       buffer.putUint8(_kFloat64List);
-      _writeSize(buffer, value.length);
+      writeSize(buffer, value.length);
       buffer.putFloat64List(value);
     } else if (value is List) {
       buffer.putUint8(_kList);
-      _writeSize(buffer, value.length);
+      writeSize(buffer, value.length);
       for (final dynamic item in value) {
-        _writeValue(buffer, item);
+        writeValue(buffer, item);
       }
     } else if (value is Map) {
       buffer.putUint8(_kMap);
-      _writeSize(buffer, value.length);
+      writeSize(buffer, value.length);
       value.forEach((dynamic key, dynamic value) {
-        _writeValue(buffer, key);
-        _writeValue(buffer, value);
+        writeValue(buffer, key);
+        writeValue(buffer, value);
       });
     } else {
-      throw new ArgumentError.value(value);
+      buffer.putUint8(_kUnknown);
+      writeUnknown(buffer, value);
     }
   }
 
-  static int _readSize(ReadBuffer buffer) {
-    final int value = buffer.getUint8();
-    if (value < 254)
-      return value;
-    else if (value == 254)
-      return buffer.getUint16();
-    else
-      return buffer.getUint32();
-  }
-
-  static dynamic _readValue(ReadBuffer buffer) {
+  /// Reads a value from [buffer] as written by [writeValue].
+  ///
+  /// This method is intended for use by subclasses overriding
+  /// [readUnknown].
+  dynamic readValue(ReadBuffer buffer) {
     if (!buffer.hasRemaining)
       throw const FormatException('Message corrupted');
     dynamic result;
```

Comment:
similarly here we could read out the first byte, then call readValue with that byte and the buffer, and they would override readValue and call super.readValue if they don't recognise the type